dvc/repo/ls.py

```diff
@@ -21,20 +30,39 @@ def ls(url, target=None, rev=None, recursive=None, outs_only=False):
         if target and not result:
             raise PathMissingError(target, repo, output_only=outs_only)
 
-        def prettify(path_info):
-            if path_info == target_path_info:
-                return path_info.name
-            return relpath(path_info, target_path_info)
+        @colorize(with_color)
+        def prettify(fs_node):
+            if fs_node.path_info == target_path_info:
+                return fs_node.path_info.name
+            return relpath(fs_node.path_info, target_path_info)
```

Comment:
I would say move the whole prettify thing not only colorization to command. Make `Repo.ls` return some structured info. I.e.:
```python
[{'path': path_info, 'is_out': True, 'is_dir': False}, ...]
```